lib/elixir/lib/exception.ex

```diff
@@ -765,6 +765,24 @@ defmodule UnicodeConversionError do
   end
 end
 
+defmodule StartLinkNameError do
+  defexception [arg: nil]
+
+  def message(exception) do
+    """
+    invalid argument type given for `name`.
+
+    Valid types are:
+      * nil
+      * atom
+      * tuple
```

Comment:
Done checking the tuple. I'm just not sure if I'm testing the `module` in `{:via, module, name}` right.